### PR TITLE
service/quote-server: add more unit tests for quote server

### PR DESCRIPTION
**Pull request by hairongchen**
Add more test cases for:

- grpc server
- tee handling functions

Signed-off-by: Hairong Chen hairong.chen@intel.com

### CODE REVIEW

File: service/quote-server/src/tee.rs [hunk 1]
```diff
@@ -63,10 +63,6 @@ fn generate_tdx_report_data(
 }
 
 fn get_tdx_quote(report_data: Option<String>, nonce: String) -> Result<String> {
-    if nonce.is_empty() {
```
Comment (Ruoyu-y) [inline service/quote-server/src/tee.rs:66]: So the nonce can be empty?

Comment (hairongchen) [reply service/quote-server/src/tee.rs:66]: yes, put all the possibility there, and the hash logic to generate tdx report data is also enhanced to allow empty nonce

Comment (Ruoyu-y) [reply service/quote-server/src/tee.rs:66]: Got it.

**Review by Ruoyu-y — APPROVED** [review]
LGTM

**Pull request merged**